### PR TITLE
Cirrus: preserve podman-server logs

### PR DESCRIPTION
**Pull request by edsantiago**
Output from podman system service, on system tests, is
being saved... it just hasn't been collected as an artifact.
Start collecting it. And, remove obsolete-unused-misleading
code that made me think it _was_ being collected.

Also: log system-service output for bud tests

Signed-off-by: Ed Santiago <santiago@redhat.com>
```release-note
None
```

### CODE REVIEW

**Review by vrothberg — APPROVED** [review]
LGTM

Thank you!

**Comment by edsantiago** [comment]
I don't think it's working (logs missing) but I'll look after meeting

**Comment by Luap99** [comment]
I see the log attached: https://cirrus-ci.com/task/5479070318723072
https://api.cirrus-ci.com/v1/artifact/task/5479070318723072/server_log/podman-server.log

**Comment by edsantiago** [comment]
Yes, but only for system tests. I thought it would work for `bud` and `int` tests too, and it's not. I don't think there's any way I can log `int`, but I think it's not too hard to log bud. Am working on it.

**Comment by edsantiago** [comment]
Yay logs: 
* [f36 root](https://api.cirrus-ci.com/v1/artifact/task/5536870445088768/server_log/podman-server.log)
* [f37 root](https://api.cirrus-ci.com/v1/artifact/task/6099820398510080/server_log/podman-server.log)
* [f37 rootless](https://api.cirrus-ci.com/v1/artifact/task/4622076770779136/server_log/podman-server.log)
* [f37-aarch64 root](https://api.cirrus-ci.com/v1/artifact/task/6662770351931392/server_log/podman-server.log)

...and now [bud](https://api.cirrus-ci.com/v1/artifact/task/6029451654332416/server_log/podman-server.log). Unfortunately, the bud log is empty apart from test names. All the other logs have stuff like "pulling images", "storing signatures". Does anyone know why there's nothing at all in the system-service logs for bud? Did I do something wrong?

File: test/buildah-bud/buildah-tests.diff
Comment (vrothberg) [inline test/buildah-bud/buildah-tests.diff:34]:
```suggestion
+        env CONTAINERS_CONF=${CONTAINERS_CONF:-$(dirname ${BASH_SOURCE})/containers.conf} $PODMAN_NATIVE system service --timeout=0 &>>${$PODMAN_SERVER_LOG:-/dev/stderr} &
```

@edsantiago could this fix the issue?  Not sure though ... bash isn't my strong suit.

**Comment by Luap99** [comment]
> ...and now [bud](https://api.cirrus-ci.com/v1/artifact/task/6029451654332416/server_log/podman-server.log). Unfortunately, the bud log is empty apart from test names. All the other logs have stuff like "pulling images", "storing signatures". Does anyone know why there's nothing at all in the system-service logs for bud? Did I do something wrong?

the log level is warning by default, so it could be very wall that are no warning/error logged. The fact the we see pull progress on the server side also sounds like a bug to me and not a feature, why would anybody want that. I assume bud test do not require any pulls because the images are already present in the store.

File: test/buildah-bud/buildah-tests.diff
Comment (edsantiago) [reply test/buildah-bud/buildah-tests.diff:34]: That's actually a syntax error.  The `${ VAR colon-minus default}` means to use stderr if `PODMAN_SERVER_LOG` is unset:
```console
$ echo ${PODMAN_SERVER_LOG:-/dev/stderr}
/dev/stderr
$ PODMAN_SERVER_LOG=/tmp/foo
$ echo ${PODMAN_SERVER_LOG:-/dev/stderr}
/tmp/foo
```

**Comment by edsantiago** [comment]
>we see pull progress on the server side also sounds like a bug to me

Oh, yeah. But it's existed so long that it's probably a feature now. 

>I assume bud test do not require any pulls because the images are already present in the store.

I haven't actually checked, but I used to be kinda-sure that some of the `bud` tests did image pulls? Like, isn't that why we see the `quay03` flake in podman bud tests?

**Comment by Luap99** [comment]
> I haven't actually checked, but I used to be kinda-sure that some of the bud tests did image pulls? Like, isn't that why we see the quay03 flake in podman bud tests?

Could be, maybe going through buildah just suppresses this output in the remote context for some reason.

If you want to make sure it works you could add --log-level=info, this might actually make more sense anyway because it would log the actual requests which will be useful for debugging.

**Comment by edsantiago** [comment]
>you could add --log-level=info,

Good idea. Tested locally, I like the results (informative, not TOO noisy). May help someone in the future. Done and pushed. Thank you.

**Comment by edsantiago** [comment]
...and here's the [podman server log for bud tests, with log-level info](https://api.cirrus-ci.com/v1/artifact/task/4605362603556864/server_log/podman-server.log)

/hold cancel

Ready to merge, if it meets team approval. Thanks for the advice.

**Pull request merged**
